### PR TITLE
meta-lxatac-bsp: barebox: update to 2024.09.0

### PR DESCRIPTION
**Pull request by hnez**
Just a routine update.

TODO before merging:

- [x] Test on hardware

### CODE REVIEW

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch [hunk 1]
```diff
@@ -707,27 +698,7 @@ index fc3a2386dbb9..0afbb1a1fa4f 100644
...
--	cs-gpios = <&gpiof 12 GPIO_ACTIVE_LOW>;
-+	cs-gpios = <&gpiof 12 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>;
- 	status = "okay";
- };
- 
-@@ -471,6 +395,10 @@ switch: switch@0 {
- 		interrupt-parent = <&gpioa>;
- 		interrupts = <6 IRQ_TYPE_EDGE_RISING>;
- 
-+		/* Reduce RGMII EMI emissions by reducing drive strength */
```
Comment (a3f) [inline meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:723]: Why is this being removed?

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch [hunk 2]
```diff
@@ -346,43 +346,6 @@ index 000000000000..5a07ad7996e6
...
-+ *   phandle, e.g.  gpios = <&gpioe 7 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
-+ *   Thix can expose a generic interface, but doesn't allow fine tuning.
-+ *
-+ * Not all Linux drivers support the second binding and barebox doesn't
-+ * support it all; It requires pinctrl drivers to expose to the core
-+ * a way to configure pull ups/pull downs, but we don't export this info.
-+ * Instead drivers are called with the device tree node in action and
-+ * instructed to apply it.
-+ *
-+ * We can add support for this binding for STM32 only as first step, but
```
Comment (a3f) [inline meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch]: Why is this being removed? v2024.08.0 doesn't support the new binding either. I [added support recently](https://github.com/barebox/barebox/commit/269dad50c259d2685249d533bc17946cca5001bf), but it is slated for inclusion in v2024.10.0.

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch [hunk 3]
```diff
@@ -604,17 +604,8 @@ index 8a34d15e9005..1fac9bb60cfb 100644
  &gpu {
  	status = "disabled";
  };
-@@ -148,7 +225,7 @@ adc@0 {
- 		compatible = "ti,lmp92064";
- 		reg = <0>;
- 
--		reset-gpios = <&gpioa 4 GPIO_ACTIVE_HIGH>;
-+		reset-gpios = <&gpioa 4 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
```
Comment (a3f) [inline meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:612]: Why is this being removed?

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch [hunk 4]
```diff
@@ -445,17 +408,14 @@ index 524d282a1db8..220d5b2c15e5 100644
...
 --- a/images/Makefile.stm32mp
 +++ b/images/Makefile.stm32mp
-@@ -35,6 +35,10 @@ $(call build_stm32mp_image, CONFIG_MACH_STM32MP15X_EV1, start_stm32mp15x_ev1, st
+@@ -34,6 +34,7 @@ $(call build_stm32mp_image, CONFIG_MACH_STM32MP15XX_DKX, start_stm32mp15xx_dkx,
+ $(call build_stm32mp_image, CONFIG_MACH_STM32MP15X_EV1, start_stm32mp15x_ev1, stm32mp15x-ev1)
  
  $(call build_stm32mp_image, CONFIG_MACH_LXA_MC1, start_stm32mp157c_lxa_mc1, stm32mp157c-lxa-mc1)
++$(call build_stm32mp_image, CONFIG_MACH_LXA_TAC, start_stm32mp15xc_lxa_tac, stm32mp15xc-lxa-tac)
  
-+pblb-$(CONFIG_MACH_LXA_TAC) += start_stm32mp15xc_lxa_tac
```
Comment (a3f) [inline meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch]: This reverts https://github.com/linux-automation/meta-lxatac/commit/e285dad4f06688c9df3f5b1d9720a7db5191542b which is likely not intended.

File: meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config [hunk 5]
```diff
@@ -18,7 +18,7 @@ image @IMAGE@fip {
     fip {
         fw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
         hw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
-        nt-fw = "barebox-stm32mp15xc-lxa-tac.img"
+        nt-fw = "barebox-stm32mp15xc-lxa-tac.stm32"
```
Comment (a3f) [inline meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config]: The .stm32 image contains at 256 byte header that I suspect will be executed as code. The .img omitted that header.

**Review by a3f — CHANGES_REQUESTED** [review]

**Comment by hnez** [comment]
It looks like I screwed up somewhere while rebasing the changes on top of 2024.08.0. I will have to investigate that a bit.

Since we do not need this update in the next stable release, I will change this to a draft PR and investigate once the release is out.

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch (hunk 3 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:612]: This change has made it into the mainline Linux https://github.com/torvalds/linux/commit/1d4e8b91d242c4afa1970ca6f7e850aab66bf452 and mainline barebox https://github.com/barebox/barebox/commit/a0991ad989b2b678aecf1ba85ee6daad469aff89 .

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch (hunk 1 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0103-ARM-stm32mp-add-Linux-Automation-TAC-Generation-3.patch:723]: This change has made it into the mainline Linux https://github.com/torvalds/linux/commit/7fd195f01ae52871b04b752447f865e1a6661487 and mainline barebox https://github.com/barebox/barebox/commit/a0991ad989b2b678aecf1ba85ee6daad469aff89 .

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch (hunk 2 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch]: It looks like the respective `fixup!` commits did not make it to the `v2024.03.0/customers/lxa/tac` branch or got force pushed away (sounds like something I would do by accident tbh).
I've made sure to push them to `v2024.03.0/customers/lxa/tac` again and have also included them in the last update of this PR.

File: meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch (hunk 4 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-ARM-stm32mp-add-Linux-Automation-TAC-board.patch]: It looks like the respective `fixup!` commit did not make it to the `v2024.03.0/customers/lxa/tac` branch or got force pushed away (sounds like something I would do by accident tbh).
I've made sure to push it to `v2024.03.0/customers/lxa/tac` again and have also included it in the last update of this PR.

File: meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config (hunk 5 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config]: My bad. This was a result of me missing the fixup that did the `stm32` --> `img` transition.
The fixup from https://github.com/linux-automation/meta-lxatac/commit/e285dad4f06688c9df3f5b1d9720a7db5191542b is now back in and this change no longer required.

**Comment by hnez** [comment]
I've revisited this PR now and did indeed find a mistake while rebasing - I was missing two fixup commits that changed the image file type and added pull-ups to the button inputs.
This should be fixed now.

While at it I also updated to 2024.09 instead of 2024.08.

**Comment by hnez** [comment]
Hi @a3f, could you have a second look at this PR?
And maybe https://github.com/linux-automation/meta-lxatac/pull/183 as well.

**Review by a3f — APPROVED** [review]

**Pull request merged**